objectbox-models.json

```diff
@@ -0,0 +1,38 @@
+{
```

Comment:
Actually, regarding your second point, this is why I decided to put the file into the root directory:

- assuming you have multiple source files in different directories, which might all define models, separate JSON files would make the act of merging much more difficult, because to make sure the merge is consistent, all other JSON files need to be read and checked regarding conflicts beforehand
- it works fine with `flutter run`. I'm actually not sure whether Flutter also compiles the Dart code like you mentioned, but it seems like at least the current working directory Flutter code is executed from is still the project's root directory

Also, I would've definitely liked if the (or one) generated `.g.dart` file(s) could contain all model definitions, but unfortunately, all of the Dart AST parsing capabilities, presumably provided by `dart:analyzer` are even less well-documented than the generator itself, so it's virtually impossible to get this to run.

Reply:
Model.json - you're right that there should be only a single file, what I'm saying is it should be in specific to a single package (tests seem to be standalone). 

Generated code - no AST parsing is necessary at all. The code is already generated right? Just generate the code directly with the information from the model.json file (during build step). 
Go does it slightly differently but just for reference: https://github.com/objectbox/objectbox-go/blob/master/examples/tasks/internal/model/task.obx.go#L63

What prevents representing a model using classes instead of a map<string, dynamic>? 
Something like [this generated code from go](https://github.com/objectbox/objectbox-go/blob/master/examples/tasks/internal/model/task.obx.go#L25) would be (probably) also necessary for queries in Dart as well, see discussion here: https://github.com/objectbox/objectbox-dart/issues/5#issuecomment-531496314.

Are we on the same page?

Reply:
As an example, the generated code could contain something like the following. 

```dart
class _TestEntity {
  PropertyLong id;
  PropertyString text;
  PropertyInt number;
}

var TestEntity_ = new _TestEntity();
TestEntity_.id = PropertyLong(id: 1, uid: 9211738071025439652, type: 6);
TestEntity_.text = PropertyString(id: 2, uid: 8804670454579230281, type: 9);
TestEntity_.number = PropertyInt(id: 3, uid: 1119420894747302346, type: 3);

```

Something like that would be necessary for Queries anyway. See discussion in  https://github.com/objectbox/objectbox-dart/issues/5

Reply:
I'm just a little sceptical about maintainability of the binding with all these maps everywhere - all IDE help with refactoring, show usage, etc. is impossible because it's just strings

Reply:
As a start, a `_getOBXModel()` could be replaced to return an instance of `Entity` that could look sth like:
```dart

class Entity {
  final int id;
  final int uid;
  final string name;
  List<Property> properties;
}
```

You get the idea what the `Property` class would look like.

Reply:
I made the respective changes to use as few maps as possible; sorry for the large commit for that, but although logically everything stayed mostly the same, it required quite a lot of code adjustments.